tests/testnoise.nim

```diff
@@ -128,12 +139,18 @@ suite "Noise":
       await sconn.write("Hello!".cstring, 6)
       await readTask
       await sconn.close()
+      await conn.close()
       await transport1.close()
+      await transport2.close()
 
       result = true
 
     check:
       waitFor(testListenerDialer()) == true
+      getTracker(AsyncStreamReaderTrackerName).isLeaked() == false
```

Comment:
use `teardown`? https://nim-lang.org/docs/unittest.html#examples

Reply:
for all the resources we open? (conns, transports etc)?

Reply:
well, I guess - every test should release everything and leave a blank slate, so the more the merrier

Reply:
Yes but that's the point of these trackers, if we forget something this check fails.

I think resource disposal is definitely part of the test btw so no need to use `teardown` as we are not really doing any `setup` (again it's part of the test to setup)

Reply:
the disposal can be part of the test (though I find the tests keep repeating the same setup/close code over and over - some refactoring here would probably yield significant simplicity savings) - checking that the disposal was done correctly could be part of the teardown or the setup (ie checking that chronos is in a "fresh" state at beginning and end of test

Reply:
I just realized teardown supports checks anyway :) my bad